daemon/api.go

```diff
@@ -222,19 +213,12 @@ func loginUser(c *Command, r *http.Request) Response {
 		return Unauthorized("cannot get discharge authorization")
 	}
 
-	authenticatedUser := userAuthState{
-		Username:   loginData.Username,
-		Macaroon:   macaroon,
-		Discharges: []string{discharge},
-	}
-	// TODO Handle better the multi-user case.
-	authStateData := authState{Users: []userAuthState{authenticatedUser}}
-
 	overlord := c.d.overlord
 	state := overlord.State()
-	state.Lock()
-	state.Set("auth", authStateData)
-	state.Unlock()
+	_, err = auth.NewUser(state, loginData.Username, macaroon, []string{discharge})
+	if err != nil {
+		return InternalError("cannot persist authentication details")
```

Comment:
Nevermind, sorry. We'll actually need to handle the error and return it as indicated. Once we read from state the error shows up.